libtenzir/builtins/operators/batch.cpp

```diff
@@ -148,18 +148,39 @@ class plugin final : public virtual operator_plugin<batch_operator>,
         .primary(timeout->source)
         .throw_();
     }
-
     return std::make_unique<batch_operator>(
       limit ? limit->inner : defaults::import::table_slice_size,
       timeout ? timeout->inner : duration::max(), event_order::ordered);
   }
 
   auto make(invocation inv, session ctx) const
     -> failure_or<operator_ptr> override {
-    argument_parser2::operator_("batch").parse(inv, ctx).ignore();
-    return std::make_unique<batch_operator>(defaults::import::table_slice_size,
-                                            duration::max(),
-                                            event_order::ordered);
+    auto limit = std::optional<located<uint64_t>>{};
+    auto timeout = std::optional<located<duration>>{};
+    argument_parser2::operator_("batch")
+      .add(limit, "<limit>")
+      .add("timeout", timeout)
+      .parse(inv, ctx)
+      .ignore();
+    auto failed = false;
+    if (limit and limit->inner == 0) {
+      diagnostic::error("batch size must not be 0")
+        .primary(limit->source)
+        .emit(ctx);
+      failed = true;
+    }
+    if (timeout and timeout->inner <= duration::zero()) {
+      diagnostic::error("timeout must be a positive duration")
+        .primary(timeout->source)
+        .emit(ctx);
+      failed = true;
```

Comment:
Wouldn't it be shorter if you didn't use the failed var at all and just returned here and above?

Reply:
Writing it as Dominik did means that multiple diagnostics can be emitted. In general this is a good practice, as we want to be as helpful as possible. Sometimes it's too awkward to write, and then it's fine to return early. But I feel like this here is okay.